File: app/src/main/java/com/example/arysu/retrofitdemo/adapter/EmployeeAdapter.java
package com.example.arysu.retrofitdemo.adapter;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.arysu.retrofitdemo.R;
import com.example.arysu.retrofitdemo.model.Employee;

import java.util.ArrayList;

public class EmployeeAdapter extends RecyclerView.Adapter<EmployeeAdapter.EmployeeViewHolder> {

    private ArrayList<Employee>dataList;

    public EmployeeAdapter(ArrayList<Employee>dataList){
        this.dataList = dataList;
    }

    @Override
    public EmployeeViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        View view = layoutInflater.inflate(R.layout.row_employee, parent, false);
        return new EmployeeViewHolder(view);
    }

    @Override
    public void onBindViewHolder(EmployeeViewHolder holder, int position) {
        holder.txtEmpName.setText(dataList.get(position).getName());
        holder.txtEmpEmail.setText(dataList.get(position).getEmail());
        holder.txtEmpPhone.setText(dataList.get(position).getPhone());
    }

    @Override
    public int getItemCount() {
        return dataList.size();
    }


    class EmployeeViewHolder extends RecyclerView.ViewHolder{
        TextView txtEmpName, txtEmpEmail, txtEmpPhone;
         EmployeeViewHolder( View itemView) {
            super(itemView);
            txtEmpName = itemView.findViewById(R.id.txt_employee_name);
            txtEmpEmail = itemView.findViewById(R.id.txt_employee_email);
            txtEmpPhone = itemView.findViewById(R.id.txt_employee_phone);
        }
    }
}

File: app/src/main/java/com/example/arysu/retrofitdemo/adapter/EmployeeAdapter.kt
package com.example.arysu.retrofitdemo.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.example.arysu.retrofitdemo.R
import com.example.arysu.retrofitdemo.model.Employee

import java.util.ArrayList

class EmployeeAdapter(private val dataList: ArrayList<Employee>) : RecyclerView.Adapter<EmployeeAdapter.EmployeeViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): EmployeeViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val view = layoutInflater.inflate(R.layout.row_employee, parent, false)
        return EmployeeViewHolder(view)
    }

    override fun onBindViewHolder(holder: EmployeeViewHolder, position: Int) {
        holder.txtEmpName.text = dataList[position].name
        holder.txtEmpEmail.text = dataList[position].email
        holder.txtEmpPhone.text = dataList[position].phone
    }

    override fun getItemCount(): Int {
        return dataList.size
    }


    inner class EmployeeViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var txtEmpName: TextView
        var txtEmpEmail: TextView
        var txtEmpPhone: TextView

        init {
            txtEmpName = itemView.findViewById(R.id.txt_employee_name)
            txtEmpEmail = itemView.findViewById(R.id.txt_employee_email)
            txtEmpPhone = itemView.findViewById(R.id.txt_employee_phone)
        }
    }
}
